src/OrleansTestKit/TestGrainIdentity.cs

```diff
@@ -44,16 +52,18 @@ public string IdentityString
 
         public int TypeCode { get { throw new NotImplementedException(); } }
 
-        public TestGrainIdentity(Guid id)
+        public TestGrainIdentity(Guid id, string keyExtension = null)
         {
             PrimaryKey = id;
             _keyType = KeyType.Guid;
+            KeyExtension = keyExtension;
```

Comment:
When keyExtension is not null, I believe this should set the KeyType to `GuidCompound`